### ISSUE TITLE
Could not connect to ProximityService

### ISSUE THREAD

**Issue by quantran141** [kontaktio/kontakt-android-sdk#182]
**Issue**
=================

![crash](https://user-images.githubusercontent.com/2345094/36965319-e7b281e4-208b-11e8-98fa-1a5c004e9ea8.PNG)

**Implement**
=================

private void initializeProximityManagerFactory() {
        kontaktManager = ProximityManagerFactory.create(LacaApplication.getInstance());
        kontaktManager.configuration()
                .scanMode(ScanMode.BALANCED)
                .scanPeriod(ScanPeriod.RANGING)
                .activityCheckConfiguration(ActivityCheckConfiguration.DEFAULT)
                .deviceUpdateCallbackInterval(TimeUnit.SECONDS.toMillis((1)))
                .eddystoneFrameTypes(EnumSet.of(EddystoneFrameType.URL))
                .rssiCalculator(RssiCalculators.newLimitedMeanRssiCalculator(5));
        kontaktManager.setIBeaconListener(new IBeaconListener() {
            @Override
            public void onIBeaconDiscovered(IBeaconDevice iBeacon, IBeaconRegion region) {
                // do discover            
            }
            @Override
            public void onIBeaconsUpdated(List<IBeaconDevice> iBeacons, IBeaconRegion region) {
              // do update beacon
            }
            @Override
            public void onIBeaconLost(IBeaconDevice iBeacon, IBeaconRegion region) {
			  // do lost beacon
            }
        });
}

private void startFindBeacon() {
	Log.d(TAG,"Start Find Beacon");
	if(!kontaktManager.isConnected()) {
		Log.d(TAG,"Connecting.... ");
		kontaktManager.connect(() -> kontaktManager.startScanning());
	}
}

**AndroidManifest.xml**
=================

![crash01](https://user-images.githubusercontent.com/2345094/36965579-a75f9efa-208c-11e8-9c8d-97a14d5c0fa5.PNG)

**Version**
=================

![crash02](https://user-images.githubusercontent.com/2345094/36965851-76ef309a-208d-11e8-8e7a-20a24e6e5d3e.PNG)

**Comment by jjimenez0611** [comment]
I have the same problem...

Could not connect to ProximityService. Please check if the ProximityService is registered in AndroidManifest.xml

 <android:name="com.kontakt.sdk.android.ble.service.ProximityService" android:exported="false"/>

**Comment by sbobowiec** [comment]
Hey. Please try our [sample app](https://github.com/kontaktio/kontakt-beacon-admin-sample-app) on the same device and let me know if there are still problems with connect to proximity manager.

**Comment by jjimenez0611** [comment]
I still have the same error this is my configuration:
KontaktSDK.initialize("xxxxxxxxxxxxxxx")
                .setDebugLoggingEnabled(BuildConfig.DEBUG)
                .setLogLevelEnabled(LogLevel.DEBUG, true)
                .setCrashlyticsLoggingEnabled(true);

In Fabric crashlytics the 99% of the errors are on 6.0.1 Android OS...

**Comment by robertbielas** [comment]
If you are interested in the code that throws this exception, here it is:
```

  private void bindService() {
    Logger.d("Binding to ProximityService...");
    final Intent serviceIntent = new Intent(context, ProximityService.class);
    final boolean isBindRequestSent = context.bindService(serviceIntent, serviceConnection, Context.BIND_AUTO_CREATE);
    if (!isBindRequestSent) {
      final String serviceName = ProximityService.class.getSimpleName();
      throw new RuntimeException(
          String.format("Could not connect to %s. Please check if the %s is registered in AndroidManifest.xml", serviceName, serviceName));
    }
  }
```

You can see that the exception is thrown if the following statement:

`      final boolean isBindRequestSent = context.bindService(serviceIntent, serviceConnection, Context.BIND_AUTO_CREATE);`

initializes isBindRequestSent flag to false.

According to: https://stackoverflow.com/questions/17780049/in-what-case-does-bindservice-return-false
there is a couple of reasons why this may be the case. First, the context is not valid anymore. E.g. it represents an Activity / Fragment that has been already destroyed. Guessing that this is not the case here, since LacaApplication.getInstance() prolly returns the app's context.

Moving further, the issue very often can be caused by omitting the service tag in the app's Manifest. But it is not this neither, since the service has been declared properly as shown in the comment above.

Some further reading yielded this:
https://coderedirect.com/questions/85220/bind-service-to-activity-in-android
with one of the comments indicating that the solution might be in starting the service in the foreground mode. 

Sdk series 5.0.xx introduced the concept of starting the ProximityService as a ForegroundService: https://knowledgebase.kontakt.io/mobile/android/qsg/usage/#scanning-problems-troubleshooting

TL;DR - you need to create a foreground notification and pass it to the sdk as described in the docs. Starting this service in the foreground mode boosts its priority, making it harder for the system to kill it - and I'm betting this is the issue here, the ProximityService not existing by the time the bind method is invoked. 

Also, do you have the phones that cause the app to crash like this? 
If you do, please add the following line:
` Logger.enableAllLoggerLevels();`
       
Since we are on the brink of 2022 and with Android 12 being available to the broader audience, maybe this ticket is not relevant to the author any longer (the issue happens on Android 6). Still, let us know if the issue persists, attaching a file with all sdk's logs unlocked by the line provided above.
